nanoFramework.WebServer/WebServer.cs

```diff
@@ -495,106 +495,81 @@ private void StartListener()
                 new Thread(() =>
                 {
                     bool isRoute = false;
-                    CallbackRoutes route;
-                    int urlParam;
-                    bool isFound;
-                    int incForSlash;
-                    string toCompare;
-                    string routeStr;
-                    string rawUrl;
+                    string rawUrl = context.Request.RawUrl;
+
+                    //This is for handling with transitory or bad requests
+                    if (rawUrl == null)
+                        return;
+
+                    int urlParam = rawUrl.IndexOf(ParamStart);
 
                     foreach (var rt in _callbackRoutes)
                     {
-                        route = (CallbackRoutes)rt;
-                        urlParam = context.Request.RawUrl.IndexOf(ParamStart);
-                        isFound = false;
-                        routeStr = route.Route;
-                        rawUrl = context.Request.RawUrl;
-                        incForSlash = routeStr.IndexOf('/') == 0 ? 0 : 1;
-                        toCompare = route.CaseSensitive ? rawUrl : rawUrl.ToLower();
-                        if (toCompare.IndexOf(routeStr) == incForSlash)
+                        CallbackRoutes route = (CallbackRoutes)rt;
+
+                        bool isFound;
```

Comment:
Reason why isFoud and all the others are declared outside of the foreach loop is for performance reasons with nanoFramework. In this loop, it will be allocated and reallocated and again and again. So allocating it as high as possible will save from this behavior and optimize the speed of execution.
So please move them back where they were. If you think what they do miss clarity, feel free to add a comment :-)